ui/src/components/TrafficPage.tsx

```diff
@@ -50,50 +51,55 @@ export const TrafficPage: React.FC<TrafficPageProps> = ({setAnalyzeStatus, onTLS
     const classes = useLayoutStyles();
 
     const [entries, setEntries] = useState([] as any);
+    const [entriesBuffer, setEntriesBuffer] = useState([] as any);
     const [focusedEntryId, setFocusedEntryId] = useState(null);
     const [selectedEntryData, setSelectedEntryData] = useState(null);
     const [connection, setConnection] = useState(ConnectionStatus.Closed);
 
     const [tappingStatus, setTappingStatus] = useState(null);
 
-    const [disableScrollList, setDisableScrollList] = useState(false);
+    const [isSnappedToBottom, setIsSnappedToBottom] = useState(true);
 
-    const [query, setQueryDefault] = useState("");
+    const [query, setQuery] = useState("");
     const [queryBackgroundColor, setQueryBackgroundColor] = useState("#f5f5f5");
+    const [addition, updateQuery] = useState("");
 
     const [queriedCurrent, setQueriedCurrent] = useState(0);
     const [queriedTotal, setQueriedTotal] = useState(0);
 
     const [startTime, setStartTime] = useState(0);
 
-    const setQuery = async (query) => {
-        if (!query) {
-            setQueryBackgroundColor("#f5f5f5")
-        } else {
-            const data = await api.validateQuery(query);
-            if (data.valid) {
-                setQueryBackgroundColor("#d2fad2")
+    useEffect(() => {
+        (async function() {
+            if (!query) {
+                setQueryBackgroundColor("#f5f5f5")
             } else {
-                setQueryBackgroundColor("#fad6dc")
+                const data = await api.validateQuery(query);
+                if (data.valid) {
+                    setQueryBackgroundColor("#d2fad2");
+                } else {
+                    setQueryBackgroundColor("#fad6dc");
+                }
             }
-        }
-        setQueryDefault(query)
-    }
+        })();
+    }, [query]);
 
-    const updateQuery = (addition) => {
+    useEffect(() => {
         if (query) {
-            setQuery(`${query} and ${addition}`)
+            setQuery(`${query} and ${addition}`);
         } else {
-            setQuery(addition)
+            setQuery(addition);
         }
-    }
+        // eslint-disable-next-line
```

Comment:
can we remove this comment? or is this a good reason to ignore the lint

Reply:
@gadotroee that's a linter error that expects developer to include `query` in the list of states that provided on:

https://github.com/up9inc/mizu/blob/426953c3500a05825eaf72a1878c9d4412348ed2/ui/src/components/TrafficPage.tsx#L94

But if we add `query` to that list, the behavior changes so we need to ignore that ESLint check.

Reply:
why it can be like before? without state and only function that passed?

```
    const updateQuery = (addition) => {
        if (query) {
            setQuery(`${query} and ${addition}`)
        } else {
            setQuery(addition)
        }
 }
 ```

Reply:
@gadotroee it's changed with https://github.com/up9inc/mizu/commit/803907de93a0671aba373dc01226bccfae81bad1

Without `useEffect`, the state of `query` sticks to its original value (at the time they passed to the components). To be able to get the latest state of `query` I turned it into `useEffect`.